icicle/src/vec_ops/vec_ops.cu

```diff
@@ -54,6 +54,31 @@ namespace vec_ops {
       if (tid >= row_size * column_size) return;
       out[(tid % row_size) * column_size + (tid / row_size)] = in[tid];
     }
+
+    template <typename E>
+    __global__ void bit_reverse_kernel(const E* input, unsigned n, unsigned shift, E* output)
+    {
+      int tid = blockIdx.x * blockDim.x + threadIdx.x;
+      // Handling arbitrary vector size
+      if (tid < n) {
+        int reversed_index = __brev(tid) >> shift;
+        output[reversed_index] = input[tid];
+      }
+    }
+    template <typename E>
+    __global__ void bit_reverse_inplace_kernel(E* input, unsigned n, unsigned shift)
```

Comment:
you could merge those two kernels like the implementation in ntt.cu is doing and simplify the bit_reverse() since it would not have to choose which one to call